airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -224,6 +224,10 @@ def _group_streams(
                     stream_state = self._connector_state_manager.get_stream_state(
                         stream_name=declarative_stream.name, namespace=declarative_stream.namespace
                     )
+                    for state_migration in declarative_stream.state_migrations:
```

Comment:
nit: Can we extract this in a method so that if we have to change it, we don't forget to change it at the other place? Something like:
```
def _get_state(self, stream):
    for state_migration in declarative_stream.state_migrations:
        if state_migration.should_migrate(stream_state):
            # The state variable is expected to be mutable but the migrate method returns an immutable mapping.
            stream_state = dict(state_migration.migrate(stream_state))
    return stream_state
```